p2p/kademlia.py

```diff
@@ -516,7 +517,12 @@ def ping(self, node: Node) -> bytes:
         return True
 
     async def bootstrap(self, bootstrap_nodes: List[Node], cancel_token: CancelToken) -> None:
-        bonded = await asyncio.gather(*[self.bond(n, cancel_token) for n in bootstrap_nodes])
+        bonded = await asyncio.gather(*(
+            self.bond(n, cancel_token)
+            for n
+            in bootstrap_nodes
+            if (n not in self.ping_callbacks and n not in self.pong_callbacks)
```

Comment:
I'm slightly concerned about these changes because if there's ever an API (and we might have one already) that does not clean up those callback lists as it should, we'll have a memory leak that will be hard to track. More importantly, though, we cold end up in a situation where a node with which we fail to bond once will keep failing forever because we think there's a bond in progress with it. 

Instead, I'd like to suggest we change `PeerPool` so that it never runs multiple discovery lookups in parallel, as that, together with the fact that `KademliaProtocol.lookup()` does not try to bond with peers more than once, should stop those `AlreadyWaiting` errors. I've just created #767 for that

Would you be ok with shelving just the changes that check whether a node is already present in a callback list and trying #767 for a while to see if it works as I expect?